lib/v8.js

```diff
@@ -247,29 +248,40 @@ Deserializer.prototype.readRawBytes = function readRawBytes(length) {
                         length);
 };
 
-/* Keep track of how to handle different ArrayBufferViews.
- * The default Serializer for Node does not use the V8 methods for serializing
- * those objects because Node's `Buffer` objects use pooled allocation in many
- * cases, and their underlying `ArrayBuffer`s would show up in the
- * serialization. Because a) those may contain sensitive data and the user
- * may not be aware of that and b) they are often much larger than the `Buffer`
- * itself, custom serialization is applied. */
-const arrayBufferViewTypes = [Int8Array, Uint8Array, Uint8ClampedArray,
-                              Int16Array, Uint16Array, Int32Array, Uint32Array,
-                              Float32Array, Float64Array, DataView];
-
-const arrayBufferViewTypeToIndex = new SafeMap();
-
-{
-  const dummy = new ArrayBuffer();
-  ArrayPrototypeForEach(arrayBufferViewTypes, (ctor, i) => {
-    const tag = ObjectPrototypeToString(new ctor(dummy));
-    arrayBufferViewTypeToIndex.set(tag, i);
-  });
+function arrayBufferViewTypeToIndex(abView) {
+  const type = ObjectPrototypeToString(abView);
+  if (type === '[object Int8Array]') return 0;
+  if (type === '[object Uint8Array]') return 1;
+  if (type === '[object Uint8ClampedArray]') return 2;
+  if (type === '[object Int16Array]') return 3;
+  if (type === '[object Uint16Array]') return 4;
+  if (type === '[object Int32Array]') return 5;
+  if (type === '[object Uint32Array]') return 6;
+  if (type === '[object Float32Array]') return 7;
+  if (type === '[object Float64Array]') return 8;
+  if (type === '[object DataView]') return 9;
+  // Index 10 is FastBuffer.
+  if (type === '[object BigInt64Array]') return 11;
+  if (type === '[object BigUint64Array]') return 12;
+  return -1;
 }
 
-const bufferConstructorIndex =
-  ArrayPrototypePush(arrayBufferViewTypes, FastBuffer) - 1;
```

Comment:
This line made it kind of ugly and complex to simply extend the array literal on line 259.

Reply:
The following would work and be simpler (even though a bit hacky):

```js
// Add "Array" as entry to arrayBufferViewTypes[10] above. Other constructors also work.
arrayBufferViewTypeToIndex.delete('[object Array]');
arrayBufferViewTypes[10] = FastBuffer;
const bufferConstructorIndex = 10;
```

Reply:
I initially did something like that to keep the diff small but I made the judgment call the open-coded approach better is going to be easier to understand and update.

Reply:
There is actually a even simpler solution:

```diff
--- a/lib/v8.js
+++ b/lib/v8.js
@@ -256,20 +256,23 @@ Deserializer.prototype.readRawBytes = function readRawBytes(length) {
  * itself, custom serialization is applied. */
 const arrayBufferViewTypes = [Int8Array, Uint8Array, Uint8ClampedArray,
                               Int16Array, Uint16Array, Int32Array, Uint32Array,
-                              Float32Array, Float64Array, DataView];
+                              Float32Array, Float64Array, DataView,
+                              FastBuffer, BigInt64Array, BigUint64Array];
 
 const arrayBufferViewTypeToIndex = new SafeMap();
 
 {
   const dummy = new ArrayBuffer();
   ArrayPrototypeForEach(arrayBufferViewTypes, (ctor, i) => {
+    if (ctor === FastBuffer) {
+      return;
+    }
     const tag = ObjectPrototypeToString(new ctor(dummy));
     arrayBufferViewTypeToIndex.set(tag, i);
   });
 }
 
-const bufferConstructorIndex =
-  ArrayPrototypePush(arrayBufferViewTypes, FastBuffer) - 1;
+const bufferConstructorIndex = 10;
```

Reply:
Sorry, I never replied. That would work but... I don't get warm fuzzies from do-this-except-when-it's-that logic. It's shorter but IMO the virtue of the open-coded approach is it's glaringly obvious what it does because it's so stupidly simple.